x-pack/plugins/security_solution/common/detection_engine/rule_management/api/rules/coverage_overview/response_schema.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import * as t from 'io-ts';
+import { NonEmptyArray } from '@kbn/securitysolution-io-ts-types';
+import { CoverageOverviewRuleActivity } from './request_schema';
+
+export type CoverageOverviewRuleData = t.TypeOf<typeof CoverageOverviewRuleData>;
+export const CoverageOverviewRuleData = t.type({
+  name: t.string,
+  activity: CoverageOverviewRuleActivity,
+});
+
+export type CoverageOverviewUnmappedRuleData = t.TypeOf<typeof CoverageOverviewUnmappedRuleData>;
+export const CoverageOverviewUnmappedRuleData = t.type({
+  id: t.string,
+  name: t.string,
+});
+
+export type CoverageOverviewResponse = t.TypeOf<typeof CoverageOverviewResponse>;
+export const CoverageOverviewResponse = t.exact(
+  t.type({
+    coverage: t.record(t.string, NonEmptyArray(t.string)),
+    rules_data: t.record(t.string, NonEmptyArray(CoverageOverviewRuleData)),
+    unmapped_rules_data: t.array(CoverageOverviewUnmappedRuleData),
+  })
+);
```

Comment:
Not sure why we need to have a separate `CoverageOverviewUnmappedRuleData` type. We could use the same `CoverageOverviewRuleData` type for unmapped rules + have an array of "unmapped" rules ids + use the same `rules_data` map to return their data in it.

Also, I see a few mistakes with the types. Here's what I'd suggest:

```ts
import * as t from 'io-ts';
import { NonEmptyString } from '@kbn/securitysolution-io-ts-types';
import { RuleObjectId } from '../../../../rule_schema';
import { CoverageOverviewRuleActivity } from './request_schema';

export type CoverageOverviewRuleData = t.TypeOf<typeof CoverageOverviewRuleData>;
export const CoverageOverviewRuleData = t.type({
  id: RuleObjectId,
  name: t.string,
  activity: CoverageOverviewRuleActivity,
});

export type MitreAttackId = t.TypeOf<typeof MitreAttackId>;
export const MitreAttackId = NonEmptyString;

export type CoverageOverviewResponse = t.TypeOf<typeof CoverageOverviewResponse>;
export const CoverageOverviewResponse = t.exact(
  t.type({
    coverage: t.record(MitreAttackId, t.array(RuleObjectId)), // for each MITRE id there could be an empty array of associated rules
    rules_not_covered: t.array(RuleObjectId), // it could be an empty array of rules w/o MITRE data, e.g. when there are no installed rules
    rules_data: t.record(RuleObjectId, CoverageOverviewRuleData),
  })
);
```

Reply:
Right, it will make the schemas definition and the code a bit simpler, I'll change as you suggest.

My idea of using a separate type `CoverageOverviewUnmappedRuleData` is based on unmapped rules don't interfere with mapped rules. So there is only one unmapped rule occurrence in the response data but mapped rules may occur  more than once.